build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/fixtures/http/HttpServerRule.groovy

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.fixtures.http
+
+import com.sun.net.httpserver.HttpExchange
+import com.sun.net.httpserver.HttpHandler
+import com.sun.net.httpserver.HttpServer
+import org.apache.commons.io.IOUtils
+import org.junit.rules.ExternalResource
+
+import java.util.function.Consumer
+
+class HttpServerRule extends ExternalResource {
+
+    private HttpServer server
+    private List<String> contexts = new ArrayList<>()
+    private int port
+
+    @Override
+    protected void before() throws Throwable {
+        // TODO revisit port range allocation
+        def socket = new ServerSocket(0)
```

Comment:
We do not have a good concept for sharing port ranges within build-tools-internal tests as we only apply the Port allocator plugin on the elasticsearch root project for now. This is not optimal, but I wonder if this is good enough here for now or will cause problems on CI until we proper address the problem. thoughts?

Reply:
Now looking into this I might just port this to our existing WireMock infrastructure 🤦